core/src/main/scala/com/yahoo/maha/core/bucketing/BucketSelector.scala

```diff
@@ -75,13 +107,35 @@ class BucketSelector(registry: Registry, bucketingConfig: BucketingConfig) exten
     }
   }
 
+  private def selectVersion(queryGenConfig: Option[QueryGenBucketingConfig], requestParams: BucketParams): Version = {
+    if (requestParams.userInfo.isInternal) {
+      Version.apply(queryGenConfig.get.internalDistribution.sample())
```

Comment:
We should treat Version instances (v0, v1, v2) as singletons.  Add new method e.g. Version.from(int): Option[Version] if it exists.  You can create a map to look up the versions in Version, e.g. val versions: Map[Int, Version] = List(v0, v1, v2).map(v -> (v.number, v)).toMap